static/app/components/overlayArrow.tsx

```diff
@@ -1,79 +1,105 @@
+import {forwardRef, ForwardRefRenderFunction, HTMLAttributes} from 'react';
 import styled from '@emotion/styled';
+import {Placement} from '@popperjs/core';
 
 import {ColorOrAlias} from 'sentry/utils/theme';
 
-type Props = {
+type Props = HTMLAttributes<HTMLDivElement> & {
   background?: ColorOrAlias;
   border?: ColorOrAlias;
-  'data-placement'?: string;
+  placement?: Placement;
   size?: number;
+  strokeWidth?: number;
 };
 
-const OverlayArrow = styled('span')<Props>`
-  pointer-events: none;
-  position: absolute;
-  width: ${p => p.size}px;
-  height: ${p => p.size}px;
+const BaseOverlayArrow: ForwardRefRenderFunction<HTMLDivElement, Props> = (
+  {
+    size = 16,
+    strokeWidth = 1,
+    placement,
+    background = 'backgroundElevated',
+    border = 'translucentBorder',
+    ...props
+  },
+  ref
+) => {
+  /**
+   * SVG height
+   */
+  const h = Math.round(size * 0.4);
+  /**
+   * SVG width
+   */
+  const w = size;
+  /**
+   * SVG stroke width
+   */
+  const s = strokeWidth;
+  const arrowPath = [
+    `M 0 ${h - s / 2}`,
+    `C ${w * 0.25} ${h - s / 2} ${w * 0.45} ${s / 2} ${w / 2} ${s / 2}`,
+    `C ${w * 0.55} ${s / 2} ${w * 0.75} ${h - s / 2} ${w} ${h - s / 2}`,
+  ].join('');
 
-  &::before,
-  &::after {
-    content: '';
-    display: block;
-    position: absolute;
-    height: ${p => p.size}px;
-    width: ${p => p.size}px;
-    border: solid ${p => p.size! / 2}px transparent;
-  }
+  return (
+    <Wrap ref={ref} placement={placement} size={size} {...props}>
```

Comment:
Do you have to make the `Wrap` a separate element? can it be the `SVG`?

Reply:
This has to do with the way popperjs calculates arrow padding. The inner SVG element is not a square – by default its dimensions are 16px by 6px. We need to add a square wrapper (16px by 16px) around it so that when we rotate the arrow sideways, popperjs leaves enough padding to avoid hitting the corners.